### PR TITLE
CreateAnomalyDetectorTool supports empty model_type

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
Same to the PPLTool, CreateAnomalyDetectorTool should also support empty model_type parameter, rather than throwing an `IllegalArgumentException`, we need to make the behavior consistent between the tools.

### Related Issues
No issue.

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/skills/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by yuye-aws — APPROVED** [review]
Looks good to me.

**Comment by yuye-aws** [comment]
[Flakey tests](https://github.com/opensearch-project/skills/actions/runs/11626676324/job/32378711529?pr=457) coming from RAG Tool IT. Can @mingshl take a look?

**Comment by yuye-aws** [comment]
@gaobinlong Can you check the comment here https://github.com/opensearch-project/skills/pull/452#discussion_r1828681195? You can update the PR after Heng's PR gets merged.

File: src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java [hunk 1]
```diff
@@ -429,7 +429,11 @@ public CreateAnomalyDetectorTool create(Map<String, Object> map) {
                 throw new IllegalArgumentException("model_id cannot be empty.");
             }
             String modelType = (String) map.getOrDefault("model_type", ModelType.CLAUDE.toString());
-            if (!ModelType.OPENAI.toString().equalsIgnoreCase(modelType) && !ModelType.CLAUDE.toString().equalsIgnoreCase(modelType)) {
+            // if model type is empty, use the default value
+            if (modelType.isEmpty()) {
```
Comment (zane-neo) [inline src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java:433]: Not understand why this change as in line 431 there's already a default value which should suggest `modelType` is not empty, isn't it?

Comment (gaobinlong) [reply src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java:433]: No, the default value takes effect only if the key `model_type` doesn't exist in the map or its value is null.

Comment (zane-neo) [reply src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java:433]: Got it, this is to handle empty string case instead of null value.

**Review by zane-neo — APPROVED** [review]
